controllers/component_image_controller.go

```diff
@@ -113,53 +107,17 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		delete(metrics.RepositoryTimesForMetrics, componentIdForMetrics)
 
 		if controllerutil.ContainsFinalizer(component, ImageRepositoryComponentFinalizer) {
-			pushRobotAccountName, pullRobotAccountName := generateRobotAccountsNames(component)
-
-			quayClient := r.BuildQuayClient(log)
-			isPushRobotAccountDeleted, err := quayClient.DeleteRobotAccount(r.QuayOrganization, pushRobotAccountName)
-			if err != nil {
-				log.Error(err, "failed to delete push robot account", l.Action, l.ActionDelete, l.Audit, "true")
-				// Do not block Component deletion if failed to delete robot account
-			}
-			if isPushRobotAccountDeleted {
-				log.Info(fmt.Sprintf("Deleted push robot account %s", pushRobotAccountName), l.Action, l.ActionDelete)
-			}
-
-			isPullRobotAccountDeleted, err := quayClient.DeleteRobotAccount(r.QuayOrganization, pullRobotAccountName)
-			if err != nil {
-				log.Error(err, "failed to delete pull robot account", l.Action, l.ActionDelete, l.Audit, "true")
-				// Do not block Component deletion if failed to delete robot account
-			}
-			if isPullRobotAccountDeleted {
-				log.Info(fmt.Sprintf("Deleted pull robot account %s", pullRobotAccountName), l.Action, l.ActionDelete)
-			}
-
-			imageRepo := generateRepositoryName(component)
-			isRepoDeleted, err := quayClient.DeleteRepository(r.QuayOrganization, imageRepo)
-			if err != nil {
-				log.Error(err, "failed to delete image repository", l.Action, l.ActionDelete, l.Audit, "true")
-				// Do not block Component deletion if failed to delete image repository
-			}
-			if isRepoDeleted {
-				log.Info(fmt.Sprintf("Deleted image repository %s", imageRepo), l.Action, l.ActionDelete)
-			}
-
-			if err := r.Client.Get(ctx, req.NamespacedName, component); err != nil {
-				log.Error(err, "failed to get Component", l.Action, l.ActionView)
-				return ctrl.Result{}, err
-			}
 			controllerutil.RemoveFinalizer(component, ImageRepositoryComponentFinalizer)
 			if err := r.Client.Update(ctx, component); err != nil {
-				log.Error(err, "failed to remove image repository finalizer", l.Action, l.ActionUpdate)
+				log.Error(err, "failed to remove image repository finalizer", l.Action, l.ActionUpdate, "componentName", component.Name)
```

Comment:
Rather than checking if all components are cleaned and then clean this code, maybe it's easier to run a migration script that just removes this finalizers? But up to you.